### PR TITLE
feat: Preference to toggle Net and Gross Sales

### LINKED ISSUE #1015: Option to toggle Gross and Net Sales
### Feature Request

**Feature Description**

Currently the user can only see Gross Sales in Dashboard. There should be an option in Settings Fragment which allow user to toggle between Gross sales and Net Sales to be shown in Dashboard.  The flow is in accordance with EventBrite app. Below is a gif of EventBrite for the same.

20180605123203-7c011ad592 gif-2-mp4 com

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by Masquerade0097**
Fixes #1015 

Checklist:

- [ ] I have checked for PMD and check-style issues 
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream development branch.

Changes: 
1. Added Preference to toggle Net and Gross Sales.

Screenshots for the change:
20180718074730-7c011ad592 gif-2-mp4 com

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/app/core/settings/SalesDataSettings.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        CheckBoxPreference grossSales = (CheckBoxPreference) findPreference("gross_sales");
+
+        Preference.OnPreferenceChangeListener listener = (preference, newValue) -> {
+            String key = preference.getKey();
+
+            switch (key) {
+                case "gross_sales":
+                    netSales.setChecked(false);
+                    break;
+                case "net_sales":
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/settings/SalesDataSettings.java]: Make constants

File: app/src/main/java/org/fossasia/openevent/app/core/settings/SalesDataSettings.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    @Override
+    @SuppressWarnings("PMD.DataflowAnomalyAnalysis") // Inevitable DU Anomaly
+    public void onCreatePreferencesFix(@Nullable Bundle bundle, String rootKey) {
+        PreferenceManager manager = getPreferenceManager();
+        manager.setSharedPreferencesName(Constants.FOSS_PREFS);
+
+        setPreferencesFromResource(R.xml.sales_data_display, rootKey);
+
+        CheckBoxPreference netSales = (CheckBoxPreference) findPreference("net_sales");
+        CheckBoxPreference grossSales = (CheckBoxPreference) findPreference("gross_sales");
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/settings/SalesDataSettings.java]: Reuse

**Comment by Masquerade0097** [comment]
@iamareebjamal @sridharjajoo Please review.

File: app/src/main/java/org/fossasia/openevent/app/core/settings/SalesDataSettings.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        listener = (preference, newValue) -> {
+            String key = preference.getKey();
+
+            switch (key) {
+                case GROSS_SALES:
+                    sales = (CheckBoxPreference) findPreference("net_sales");
+                    sales.setChecked(false);
+                    break;
+                case NET_SALES:
+                    sales = (CheckBoxPreference) findPreference("gross_sales");
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/settings/SalesDataSettings.java]: Reuse

File: app/src/main/java/org/fossasia/openevent/app/core/settings/SalesDataSettings.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                    sales.setChecked(false);
+                    break;
+                default:
+                    break;
+            }
+            return (Boolean) newValue;
+        };
+
+        setPreferenceChangeListener("net_sales");
+        setPreferenceChangeListener("gross_sales");
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/settings/SalesDataSettings.java]: Reuse

File: app/src/main/res/xml/preferences.xml [hunk 5]
```diff
@@ -19,6 +19,10 @@
         android:title="Device Name"
         android:summary="@string/device_name_summary"/>
 
+    <Preference
+        android:key="sales_data_display"
+        android:title="Sales Data Display" />
```
Comment (mishuvs) [inline app/src/main/res/xml/preferences.xml]: Add summary here

File: app/src/main/java/org/fossasia/openevent/app/core/settings/SettingsFragment.java [hunk 6]
```diff
@@ -37,6 +38,15 @@ public void onCreatePreferencesFix(@Nullable Bundle bundle, String rootKey) {
 
         setPreferencesFromResource(R.xml.preferences, rootKey);
 
+        findPreference("sales_data_display").setOnPreferenceClickListener(preference -> {
+            FragmentTransaction transaction = getFragmentManager().beginTransaction();
+            transaction
+                .replace(R.id.fragment_container, SalesDataSettings.newInstance())
```
Comment (mishuvs) [inline app/src/main/java/org/fossasia/openevent/app/core/settings/SettingsFragment.java]: Move to same line

File: app/src/main/java/org/fossasia/openevent/app/core/settings/SalesDataSettings.java (hunk 3 above)
Comment (Masquerade0097) [reply app/src/main/java/org/fossasia/openevent/app/core/settings/SalesDataSettings.java]: Sorry, I didn't get what you mean.

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/app/core/settings/SalesDataSettings.java]: `"gross_sales"` is already defined as constant

**Comment by Masquerade0097** [comment]
@mishuvs I've made the changes. Now sales summary is added by getting value from sales checkbox preference. 

@iamareebjamal @sridharjajoo @srv-twry Please review.

File: app/src/main/java/org/fossasia/openevent/app/core/settings/SettingsFragment.java [hunk 7]
```diff
@@ -20,6 +21,9 @@
 public class SettingsFragment extends PreferenceFragmentCompat {
 
     private static final String VERSION = "Version";
+    private static final String GROSS_SALES = "Gross Sales";
+    private static final String NET_SALES = "Net Sales";
+    private PreferenceManager manager;
```
Comment (mishuvs) [inline app/src/main/java/org/fossasia/openevent/app/core/settings/SettingsFragment.java]: Can be final?

File: app/src/main/java/org/fossasia/openevent/app/core/settings/SettingsFragment.java [hunk 8]
```diff
@@ -34,11 +38,21 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
     public void onCreatePreferencesFix(@Nullable Bundle bundle, String rootKey) {
-        PreferenceManager manager = getPreferenceManager();
+        manager = getPreferenceManager();
         manager.setSharedPreferencesName(Constants.FOSS_PREFS);
 
         setPreferencesFromResource(R.xml.preferences, rootKey);
 
+        findPreference(getString(R.string.sales_data_display_key)).setOnPreferenceClickListener(preference -> {
+            FragmentTransaction transaction = getFragmentManager().beginTransaction();
+            transaction.replace(R.id.fragment_container, SalesDataSettings.newInstance())
```
Comment (mishuvs) [inline app/src/main/java/org/fossasia/openevent/app/core/settings/SettingsFragment.java]: `transaction` not needed

File: app/src/main/java/org/fossasia/openevent/app/core/settings/SettingsFragment.java [hunk 9]
```diff
@@ -34,11 +38,21 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
+                .addToBackStack(null)
+                .commit();
+            return true;
+        });
+
+        setSalesDataSummary();
```
Comment (mishuvs) [inline app/src/main/java/org/fossasia/openevent/app/core/settings/SettingsFragment.java]: redundant

File: app/src/main/java/org/fossasia/openevent/app/core/settings/SettingsFragment.java [hunk 10]
```diff
@@ -34,11 +38,21 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
+
         findPreference(getString(R.string.app_version_key)).setTitle(VERSION + " " + BuildConfig.VERSION_NAME);
 
         findPreference("rate_us").setOnPreferenceClickListener(preference -> {
```
Comment (mishuvs) [inline app/src/main/java/org/fossasia/openevent/app/core/settings/SettingsFragment.java]: getString

File: app/src/main/java/org/fossasia/openevent/app/core/settings/SettingsFragment.java [hunk 11]
```diff
@@ -54,9 +68,22 @@ public void onCreatePreferencesFix(@Nullable Bundle bundle, String rootKey) {
         });
     }
 
+    public void setSalesDataSummary() {
+        String sales_data;
+
+        if (manager.getSharedPreferences().getBoolean("gross_sales", true)) {
```
Comment (mishuvs) [inline app/src/main/java/org/fossasia/openevent/app/core/settings/SettingsFragment.java]: getString

File: app/src/main/java/org/fossasia/openevent/app/core/settings/SettingsFragment.java [hunk 12]
```diff
@@ -54,9 +68,22 @@ public void onCreatePreferencesFix(@Nullable Bundle bundle, String rootKey) {
...
+            sales_data = GROSS_SALES;
+        }else {
```
Comment (mishuvs) [inline app/src/main/java/org/fossasia/openevent/app/core/settings/SettingsFragment.java]: space

File: app/src/main/res/xml/preferences.xml [hunk 13]
```diff
@@ -19,6 +19,10 @@
         android:title="Device Name"
         android:summary="@string/device_name_summary"/>
 
+    <Preference
+        android:key="@string/sales_data_display_key"
+        android:title="@string/sales_data_display" />
+
     <PreferenceCategory android:title="@string/about">
         <Preference
             android:key="rate_us"
```
Comment (mishuvs) [inline app/src/main/res/xml/preferences.xml]: string resource

File: app/src/main/res/xml/sales_data_display.xml [hunk 14]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android">
+
+    <CheckBoxPreference
+        android:key="gross_sales"
```
Comment (mishuvs) [inline app/src/main/res/xml/sales_data_display.xml]: string resource

File: app/src/main/res/xml/sales_data_display.xml [hunk 15]
```diff
@@ -0,0 +1,13 @@
...
+        android:title="@string/gross_sales"
+        android:defaultValue="false" />
+
+    <CheckBoxPreference
+        android:key="net_sales"
```
Comment (mishuvs) [inline app/src/main/res/xml/sales_data_display.xml]: same

File: app/src/main/res/xml/sales_data_display.xml [hunk 16]
```diff
@@ -0,0 +1,13 @@
...
+        android:title="Net Sales"
```
Comment (mishuvs) [inline app/src/main/res/xml/sales_data_display.xml]: same

File: app/src/main/java/org/fossasia/openevent/app/core/settings/SettingsFragment.java (hunk 10 above)
Comment (Masquerade0097) [reply app/src/main/java/org/fossasia/openevent/app/core/settings/SettingsFragment.java]: Not added in this PR.

File: app/src/main/res/xml/preferences.xml (hunk 13 above)
Comment (Masquerade0097) [reply app/src/main/res/xml/preferences.xml]: Not added in this PR

File: app/src/main/java/org/fossasia/openevent/app/core/settings/SettingsFragment.java (hunk 7 above)
Comment (Masquerade0097) [reply app/src/main/java/org/fossasia/openevent/app/core/settings/SettingsFragment.java]: It is initialized in onCreatePreferencesFix. Hence, cannot be final.

**Comment by Masquerade0097** [comment]
@mishuvs Please review.

**Pull request merged**